### PR TITLE
Fix validator waiting on start

### LINKED ISSUE #414: Fix validator waiting on start
Currently validator has some recursive method call to wait for chain start.

Solution:
- modify rpc/rest client to be event emmiter
   - rest api will poll x seconds for chain status and on start emit event
- validator should listen on start event

### PR DESCRIPTION
**Pull request by twoeths**
## Goal
+ Close #414 
## Unit test
```
Validator
2019-10-03 16:27:09  [VALIDATOR]        info: Setting up validator client...
2019-10-03 16:27:09  [VALIDATOR]        info: Setting up RPC connection...
    ✓ Should be able to connect with the beacon chain
2019-10-03 16:27:09  [VALIDATOR]        info: RPC connection successfully established: inmemory!
2019-10-03 16:27:09  [VALIDATOR]        info: Checking if chain has started...


  298 passing (21s)
  1 pending

2019-10-03 16:27:10  [VALIDATOR]        info: Chain start has occured!
```

### CODE REVIEW

File: packages/lodestar/src/validator/rpc/abstract.ts [hunk 1]
```diff
@@ -35,6 +38,25 @@ export abstract class AbstractRpcClient implements IRpcClient {
...
     this.running = false;
+    if(this.beaconNodeInterval) {
+      clearInterval(this.beaconNodeInterval);
+    }
+  }
+
+  // poll BeaconNode until it starts then inform validator
+  public waitForChainLived(): void {
+    this.beaconNodeInterval = setInterval(this.pollBeaconNode.bind(this), 1000);
+  }
```
Comment (mpetrunic) [inline packages/lodestar/src/validator/rpc/abstract.ts]: I don't think you need this method, you can start interval in "connect" method

File: packages/lodestar/src/validator/rpc/interface.ts [hunk 2]
```diff
@@ -9,7 +11,13 @@ export interface INewEpochCallback {
   (slot: Epoch): void;
 }
 
-export interface IRpcClient {
+export interface IRpcClientEvents {
+  chainLived: () => void;
```
Comment (mpetrunic) [inline packages/lodestar/src/validator/rpc/interface.ts]: I would call this event "beaconChainStarted"

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/lodestar/src/validator/validator.ts [hunk 3]
```diff
@@ -129,32 +127,11 @@ class Validator {
   private async setupRPC(): Promise<void> {
     this.logger.info("Setting up RPC connection...");
     await this.apiClient.connect();
+    this.logger.info("Checking if chain has started...");
+    this.apiClient.once("beaconChainStarted", this.run.bind(this));
```
Comment (mpetrunic) [inline packages/lodestar/src/validator/validator.ts]: I would put this in start method, it makes more sense

**Review by mpetrunic — CHANGES_REQUESTED** [review]
lgtm, just one small thing

**Review by mpetrunic — APPROVED** [review]

**Comment by mpetrunic** [comment]
@GregTheGreek labeler?

**Comment by GregTheGreek** [comment]
Fails on forks unfortunately

**Review by GregTheGreek — APPROVED** [review]

**Pull request merged**